src/response.rs

```diff
@@ -124,46 +123,52 @@ impl<'a, 'b> Response<'a, 'b> {
     ///     response.render("examples/assets/template.tpl", &data);
     /// }
     /// ```
-    pub fn render<'c, T: Encodable<Encoder<'c>, Error>>
+    pub fn render<'c, T: Encodable>
         (&mut self, path: &'static str, data: &T) {
             // Fast path doesn't need writer lock
-            match self.templates.read().get(&path) {
-                Some(t) => {
-                    let _ = t.render(self.origin, data);
-                    return
+            match self.templates.read() {
+                Ok(hm) => match hm.get(&path) {
+                    Some(t) => {
+                        let _ = t.render(self.origin, data);
+                        return
+                    },
+                    None => {}
                 },
-                None => {}
+                Err(_) => {}
```

Comment:
It says in the documentation that this only occurs if the writer panics while an exclusive lock, in that case we're screwed and might as well panic here too. Unwrap might not be so bad.

@rgs1 I'm sorry I made you change the `unwrap()` calls yesterday, I didn't read carefully enough last night.